site/content/docs/Advanced/Ransomware Protection/_index.md

```diff
@@ -0,0 +1,82 @@
+      * B2_APPLICATION_KEY_ID=xxxxxxxxxxxxxxxxxxxxxxxxx
+      * B2_APPLICATION_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+    * Run: `b2 create-key --bucket <bucket-name> <key-name> listBuckets,readBuckets,listFiles,readFiles,writeFiles,readBucketEncryption,readBucketReplications,readBucketRetentions,readFileRetentions,writeFileRetentions,readFileLegalHolds`
+    * The resulting string contains the restricted application key and password...you will need this for Kopia
+    * Disconnect and reconnect your existing Kopia repo using the new key (or create a new bucket using this key)
+    * Re-generate your API master API key (to deactivate the one you just used)
+    * Delete any existing Application keys that do not have restricted access
+
+### Even more protection
+
+So far, we have secured your access such that even if a bad actor gets access to your Kopia configuration, they can't do irreparable harm to your cloud backup.  However, what if they get access to your login credentials?  Your login credentials provide the ability to delete your data and even your entire buckets for all the buckets in your account.  But the cloud providers have protection from that too.  AWS (and S3 compatible providers) offer a feature called 'Object Locking'.  An Object Lock is applied when a file is created (or on an existing file), and it provides a specific retention date.  Until that retention date occurs, there is no way to delete the locked file.  Even using your login credentials, the file is protected from deletion.  It can still be overwritten with a new version or hidden such that it doesn't appear in a file list.  But it will always be accessible until its retention date occurs.  While Kopia supports applying Object Locks, there are some caveats:
+
+  * Object Locks must be enabled when a bucket is created.  If you already have backups in the cloud, you will need to create a new bucket with Object Locks turned on.
+  * You must enable Object Lock extension in Kopia.  By default, Kopia does not renew Object Lock retention dates, however this can be enabled in the `full-maintenance` options.  You must ensure that you run full maintenance at least as frequently as your Object Lock period to ensure Object Locks do not expire, otherwise the Retention date will pass, and your data will no longer be protected by an Object Locks.
+  * Kopia currently only supports object locks when using an S3 repo.  If you use Backblaze, you will need to use the S3 repo method with Kopia instead of the Backblaze native API.
+
+### How to enable Object Locks
+
+  * Create a new bucket with Object Locks enabled
+    * There is no need to set a default Retention time
+  * Optional: Create a restricted access key as instructed above for additional protection
+  * Create a new Kopia repository
+    * Run: `kopia repo create s3 --bucket <bucket name> --access-key=<access key> --secret-access-key=<secret access key> --retention-mode COMPLIANCE --retention-period <retention time>`
```

Comment:
it may be worth linking to the S3 documentation on retention modes so that folks aren't surprised by the restrictions on compliance mode